src/compiler.cpp

```diff
@@ -1,69 +1,80 @@
 #include <fstream>
 #include <iostream>
 
-#include "cli.h"
+#include "cli.hpp"
 #include "ast.hpp"
 
 using ast::NodePtr;
 
-NodePtr Parse(const CommandLineArguments& args);
+// Wrapper for ParseAST defined in YACC
+NodePtr Parse(const std::string& compile_source_path);
```

Comment:
This might be a bit too *extra*, but it might be worth considering using [`std::filesystem::path`](https://en.cppreference.com/w/cpp/filesystem/path) rather than strings? This has the benefit of making it clear that this variable represents a file path from its _type_ rather than its name, which is usually an improvement imo - but if you think it's not worth the complexity / overhead for students happy for you to keep it as is.

Reply:
Another interesting choice between teaching modern C++ vs not scaring students too much. I can see both points being made here and I'm slightly in favor of including this change as it doesn't look too complex and should be understandable from the type name itself. Any strong opinions @Jpnock?